ocaml/xapi/cli_progress_bar.ml

```diff
@@ -62,18 +62,16 @@ module Make(T: Floatable) = struct
 
   let string_of_bar t =
     let w = bar_width t t.current_value in
-    let line = Bytes.of_string t.line in
-    Bytes.set line 1 spinner.(t.spin_index);
+    Bytes.set t.line 1 spinner.(t.spin_index);
     t.spin_index <- (t.spin_index + 1) mod (Array.length spinner);
     for i = 0 to w - 1 do
-      Bytes.set line (prefix + i) (if i = w - 1 then '>' else '#')
+      Bytes.set t.line (prefix + i) (if i = w - 1 then '>' else '#')
     done;
-    t.line <- Bytes.unsafe_to_string line;
     let percent = Printf.sprintf "%3d" (percent t) in
     String.blit percent 0 t.line (t.width - 19) 3;
     let eta = eta t in
     String.blit eta 0 t.line (t.width - 10) (String.length eta);
-    t.line
+    Bytes.to_string t.line
```

Comment:
All these overwrites of `line.t` are fine, because we always overwrite the same area, except for the progress bar which moves forward, so that's also fine